mods/doors/init.lua

```diff
@@ -108,6 +108,33 @@ function doors.register_door(name, def)
 		end
 	end
 
+	local function check_and_blast(pos, name)
+		local node = minetest.get_node(pos)
+		if node.name == name then
+			minetest.remove_node(pos)
+		end
+	end
+
+	local function make_on_blast(base_name, door_type, other_door_type)
+		if def.only_placer_can_open then
+			return function() end
+		else
+			if door_type == "_b_1" or door_type == "_b_2" then
+				return function(pos, intensity)
+					check_and_blast(pos, name..door_type)
+					pos.y = pos.y + 1
+					check_and_blast(pos, name..other_door_type)
+				end
+			elseif door_type == "_t_1" or door_type == "_t_2" then
+				return function(pos, intensity)
+					check_and_blast(pos, name..door_type)
+					pos.y = pos.y - 1
+					check_and_blast(pos, name..other_door_type)
+				end
+			end
```

Comment:
This should use `elseif` rather than an `else if then end` chain.
Also, you can add a `step` value of `1` or `-1` instead of copy-pasting the function.